### PR TITLE
Futres trade additions

### PR DESCRIPTION
**Pull request by brossetti1**
adding pnl and fees to `FuturesTrade`

### CODE REVIEW

File: subgraphs/main.graphql [hunk 1]
```diff
@@ -109,3 +111,47 @@ type FuturesOrder @entity {
...
+  currencyKey: Bytes!
+  " currencyKey expressed as a string "
+  synth: String!
+  " the rate recorded at this timestamp "
+  rate: BigDecimal!
+  " the block which this rate was recorded "
+  block: BigInt!
+  " timestamp of the block in which the rate was recorded "
+  timestamp: BigInt!
+}
```
Comment (brossetti1) [inline subgraphs/main.graphql:157]: not sure why this is apart of my `npm run build` but maybe it should stay

Comment (Tburm) [reply subgraphs/main.graphql:157]: Yeah this is expected, the main stuff is generated during the build so if you added types it will change this file.

Comment (Tburm) [reply subgraphs/main.graphql:157]: Makes no difference on the actual deploy though, as it's built by CI.

File: src/futures.ts [hunk 2]
```diff
@@ -29,6 +29,24 @@ let ETHER = BigInt.fromI32(10).pow(18);
...
+    (positionEntity.size.lt(ZERO) && positionEntity.size.minus(event.params.size).lt(positionEntity.size))
+  ) {
+    // if so, cap the trade size at closing the position
+    tradeSize = positionEntity.size;
+  } else {
+    // otherwise calculate the difference in position size
+    tradeSize = positionEntity.size.minus(event.params.size);
+  }
+
+  return tradeSize;
```
Comment (brossetti1) [inline src/futures.ts:47]: i sort of feel like this logic is better just duplicated in both places for now - theres some refactoring that can be done later to consolidate certain logic so the functions of these events are more easily understood

Comment (brossetti1) [reply src/futures.ts:47]: just easier to read without skipping around - i have some other ideas on how to clean up the graph functions so its easier to interpret.

i think theres a better solution as far as readability, duplication isnt bad in my opinion if its easier to read by duplicating, vs hiding awayy in a function.. anyways, this topic is not really that important right now

**Pull request merged**